### PR TITLE
fix #432: Fix the regression present in v0.33

### LINKED ISSUE #432: regression in v0.0.33
### run

```
ssh_scan --target 67.205.145.156 --port 49154
```

### expect

complete without failure

### actual

v0.0.32 succeeds as expected

v0.0.33 fails with:

```
/app/lib/ssh_scan/scan_engine.rb:138:in `scan_target': undefined method `split' for nil:NilClass (NoMethodError)
	from /app/lib/ssh_scan/scan_engine.rb:191:in `block (2 levels) in scan'
```

## Diff between v0.0.32 (good) and v0.0.33 (bad)

```
$ git log -p -U0 --no-merges 0.0.32..0.0.33
commit 3a70e66a5c0d8106cad47682f090bd1ac189032f (origin/bump_for_33)
Author: Jonathan Claudius <jclaudius@mozilla.com>
Date:   Tue Feb 13 10:55:44 2018 -0500

    Bump for 33

diff --git a/lib/ssh_scan/version.rb b/lib/ssh_scan/version.rb
index f10a875..ee5244f 100644
--- a/lib/ssh_scan/version.rb
+++ b/lib/ssh_scan/version.rb
@@ -2 +2 @@ module SSHScan
-  VERSION = '0.0.32'
+  VERSION = '0.0.33'

commit a07bebe12d72f5cca383a23d446d66802f7e335c
Author: Jinank Jain <jinank94@gmail.com>
Date:   Tue Feb 13 11:44:54 2018 +0100

    fixes #429: Add the missing port argument to ssh_keyscan

diff --git a/lib/ssh_scan/scan_engine.rb b/lib/ssh_scan/scan_engine.rb
index 1cdfb68..aadbc86 100644
--- a/lib/ssh_scan/scan_engine.rb
+++ b/lib/ssh_scan/scan_engine.rb
@@ -127 +127 @@ module SSHScan
-          stdin, stdout, stderr, wait_thr = Open3.popen3('ssh-keyscan', '-t', 'rsa,dsa', target)
+          stdin, stdout, stderr, wait_thr = Open3.popen3('ssh-keyscan', '-t', 'rsa,dsa', '-p', port.to_s, target)
```

## reproducers

My [git repo](https://github.com/jumanjihouse/docker-ssh-scan) builds a docker image that contains `ssh_scan` at a specific version. The [test harness](https://github.com/jumanjihouse/docker-ssh-scan/blob/master/ci/test) runs `ssh_scan` against a live sshd with [a few options](https://github.com/jumanjihouse/docker-ssh-scan/blob/0a00889a9407bba961476b1e8aa9e772754955a1/docker-compose.yaml#L30-L32).

Currently:

* my master branch is at ssh_scan v0.0.32
* my upgrade branch is at ssh_scan v0.0.33

### reproduce in a local dev environment with these commands

```
git checkout <branch>
ci/build && ci/test
```

#### success

https://github.com/jumanjihouse/docker-ssh-scan/tree/9376f7cc519bc324bcea964a7e4ec4a3b7e78cb5 with console output [on circleci](https://circleci.com/gh/jumanjihouse/docker-ssh-scan/40)

#### failure

https://github.com/jumanjihouse/docker-ssh-scan/pull/15 with console output [on circleci](https://circleci.com/gh/jumanjihouse/docker-ssh-scan/41?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link)

## Related links

* https://github.com/jinankjain/ssh_scan/commit/7d19fa530e7ea0b4434652a623b3c0e16db33f92
* https://github.com/mozilla/ssh_scan/issues/429
* https://github.com/mozilla/ssh_scan/pull/430

### PR DESCRIPTION
**Pull request by jinankjain**
ssh_keyscan is behaving weirdly when a port is passed to it as an
argument. By weird what I mean is instead of sending to output stdout it is
redirected to stderr. This is a temporary fix and it needs more
investigation why is that the case.

@claudijd  @jumanjiman

### CODE REVIEW

**Comment by jumanjiman** [comment]
I confirm that this fixes the regression as reported in #432

**Comment by jinankjain** [comment]
@jumanjiman Thanks for confirming

**Comment by claudijd** [comment]
@jumanjiman thanks for reporting and @jinankjain for fixing!  I wasn't aware of https://github.com/jumanjihouse/docker-ssh-scan either, cool!  

@jumanjiman any interest in combining forces with ssh_scan and docker-ssh-scan?  We currently provide ssh_scan as a docker image, but maybe it's not exactly what you want.  What's the feature gap that ssh_scan isn't providing right now? I'm really open to suggestions, ideas, opinions, etc.

**Comment by jinankjain** [comment]
@claudijd Are you aware about any reason why it was redirected to stderr instead of stdout?

**Comment by claudijd** [comment]
@jinankjain I have not had time to investigate further, but my assumption is that this is ssk-key behavior, I'm mostly focused on getting the work around out to fix the issue and I will revisit to address root cause.  We may end up reporting upstream to them if that is truly the behavior.

**Comment by claudijd** [comment]
work-around deployed via gem here => https://github.com/mozilla/ssh_scan/releases/tag/0.0.34

**Comment by claudijd** [comment]
Docker hub is also updated here => https://hub.docker.com/r/mozilla/ssh_scan/

**Comment by claudijd** [comment]
I'm still not sold about whether this is an ssh-key issue as much as it's an issue with the provider in this case (digital ocean) seeing the behavior as malicious and blocking it and that combined with ssh-keyscan being a black box (ie. we call out to it as a dependency) we don't have a lot of visibility here.

**Comment by claudijd** [comment]
I started with a few tests like so...

$ ssh-keyscan -t rsa,dsa -p 49154 67.205.145.156 > output1.txt
# 67.205.145.156:49154 SSH-2.0-OpenSSH_7.6
# 67.205.145.156:49154 SSH-2.0-OpenSSH_7.6
$ ssh-keyscan -t rsa,dsa 67.205.145.156 > output2.txt
# 67.205.145.156:22 SSH-2.0-OpenSSH_7.6
# 67.205.145.156:22 SSH-2.0-OpenSSH_7.6

$ cat output1.txt 
$ cat output2.txt 
67.205.145.156 ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDN+8qiCWCtry8KuY6MvWMhcPt58co/HhvZfiduZ+UnOnatf8lirP3A3d0wyjS1R/+7XfeYxwjs2OahHKAJyyK02FYSMFbAmW8pnkCzP94LoosRUy50bZh00W6WD6m/H7bwK1QzhAN8RKT8cPKtyROTtt2Um58WXsxrIWZ+FpnaDqACFasgX5cZAQkOHK3YWhDHEP7lJOex7FHia/vz5gMJxb9B8ZfOE04bvFEIlBswiE1hy/1mXU47uUwU/qmY+uxf5dwg9Gapil1lWH7DQaQ0CHbFOUA/UCis11LzxWds4NNvFztws+rFSkE2tLVS1HE2TCwv+ljgLUf9wXiiicWH

But, after a few attempts, I'm now getting way less useful responses, which tells me maybe we're dealing with an active IPS/IDS and these requests are timing out now...

$ ssh-keyscan -t rsa,dsa -p 22 67.205.145.156
$ ssh-keyscan -t rsa,dsa 67.205.145.156

So, in short, I believe this is somewhat outside the context of the scanner, but the server target behavior.  If we can handle that behavior more nicely, like we do in the patch, that is great, but if it's going to timeout we're just going to need to expect that scenario and report it as such.
